nautobot/core/models/fields.py

```diff
@@ -415,3 +417,48 @@ def formfield(self, form_class=fields.DynamicModelMultipleChoiceField, **kwargs)
         kwargs.setdefault("required", False)
         kwargs.setdefault("query_params", {"content_types": self.model._meta.label_lower})
         return super().formfield(form_class=form_class, **kwargs)
+
+
+class PositiveRangeNumberTextField(models.TextField):
+    default_error_messages = {
+        "invalid": "Invalid value. Specify a value using positive integers in a range format (i.e. '10-20').",
+    }
+
+    description = "A text based representation of positive number range."
+
+    def __init__(self, min_boundary=None, max_boundary=None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.min_boundary = min_boundary
+        self.max_boundary = max_boundary
+
+    def to_python(self, value):
+        if value is None:
+            return None
+
+        try:
+            self.expanded = sorted(parse_numeric_range(value))
+        except (ValueError, AttributeError):
+            raise exceptions.ValidationError(
+                self.error_messages["invalid"],
+                code="invalid",
+                params={"value": value},
+            )
+
+        converted_ranges = compress_range(self.expanded)
+        normalized_range = ",".join([f"{x[0]}" if x[0] == x[1] else f"{x[0]}-{x[1]}" for x in converted_ranges])
+
+        return normalized_range
+
+    def validate(self, value, model_instance):
+        """
+        Validate `value` and raise ValidationError if necessary.
+        """
+        super().validate(value, model_instance)
+
+        if self.min_boundary and self.max_boundary:
+            if self.expanded[0] < self.min_boundary or self.expanded[-1] > self.max_boundary:
+                raise exceptions.ValidationError(
+                    message=f"Invalid value. Specify a range value between {self.min_boundary}-{self.max_boundary}",
+                    code="outofrange",
+                    params={"value": value},
+                )
```

Comment:
Should we support specifying one boundary but not the other? At the very least I'd think the default behavior would be to have a `min_boundary` of 0 but no `max_boundary`?

Reply:
Added 👍